### PR TITLE
Vol2 gp

### PR DESCRIPTION
**Pull request by andrewgordonwilson**
*Description of changes:*

Added Gaussian process chapter, including notebooks for the index, priors, inference, and advanced topics. The index, priors, and inference notebooks are still being refined, but the basic structure and content are essentially complete.

### CODE REVIEW

File: chapter_gaussian-processes/gp-inference.md [hunk 1]
```diff
@@ -60,16 +60,15 @@ There are some key points to note about the predictive distributions for Gaussia
...
+- The key computational bottlenecks come from solving a linear system and computing a log determinant over an $n \times n$ symmetric positive definite matrix $K(X,X)$ for $n$ training points. Naively, these operations each incur $\mathcal{O}(n^3)$ computations, as well as $\mathcal{O}(n^2)$ storage for each entry of the kernel (covariance) matrix, often starting with a Cholesky decomposition. Historically, these bottlenecks have limited GPs to problems with fewer than about 10,000 training points, and have given GPs a reputation for being slow that has been inaccurate now for almost a decade. In advanced topics, we will discuss how GPs can be scaled to problems with millions of points.
```
Comment (astonzhang) [inline chapter_gaussian-processes/gp-inference.md:63]: Please don't remove them and just use common double quotes: "xxx".

**Comment by cheungdaven** [comment]
@d2l-bot please rebuild

**Comment by astonzhang** [comment]
@andrewgordonwilson Hi Andrew, could you try rebasing your PR to the HEAD of master to pass the CI? @AnirudhDagar may help you with this. Thanks.

**Review by AnirudhDagar — COMMENTED** [review]
Hi @andrewgordonwilson, i can see that this PR is WIP, but @astonzhang asked me to share the first code review asap. I've made a few small comments.

nit: The images rendered in gp intro can be rearranged in a more compact way. Right now it is a sequence of images centred on the page. Also, all the images need to be in an SVG format.

Ps. gp-numerics notebook is .ipynb file, so I couldn't review that but I guess that you will be converting it to .md file later.

File: chapter_gaussian-processes/gp-intro.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+Let's get a feel for how Gaussian processes operate, by starting with some examples.
+
+Suppose we observe the following dataset, of regression targets (outputs), y, indexed by inputs, x. As an example, the targets could be changes in carbon dioxide concentrations, and the inputs could be the times at which these targets have been recorded. What are some features of the data? How quickly does it seem to varying? Do we have data points collected at regular intervals, or are there missing inputs? How would you imagine filling in the missing regions, or forecasting up until x=25?
+
+![data](https://user-images.githubusercontent.com/6753639/178247765-650772fb-2622-42d0-8eff-316dc835816f.png)
```
[the remark this answers is not in the document]
Comment (astonzhang) [reply chapter_gaussian-processes/gp-intro.md:9]: [CAPTION](../img/IMG_NAME)

File: chapter_gaussian-processes/gp-inference.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+$\mathbf{f}_* | \mathbf{y}, X, X_* \sim \mathcal{N}(m_*,S_*)$, where $m_* = K(X_*,X)[K(X,X)+\sigma^2I]^{-1}\textbf{y}$, and $S = K(X_*,X_*) - K(X_*,X)[K(X,X)+\sigma^2I]^{-1}K(X,X_*)$.
+
+Typically, we do not need to make use of the full predictive covariance matrix $S$, and instead use the diagonal of $S$ for uncertainty about each prediction. Often for this reason we write the predictive distribution for a single test point $x_*$, rather than a collection of test points. 
+
+The kernel matrix has parameters $\theta$ that we also wish to estimate, such the amplitude $a$ and lengthscale $\ell$ of the RBF kernel above. For these purposes we use the _marginal likelihood_, $p(\textbf{y} | \theta, X)$, which we already derived in working out the marginal distributions to find the joint distribution over $\textbf{y},\textbf{f}_*$. As we will see, the marginal likelihood compartmentalizes into model fit and model complexity terms, and automatically encodes a notion of Occam's razor for learning hyperparameters. For a full discussion, see MacKay Ch. 28, and Rasmussen and Williams Ch. 5 (! Add references).
```
Comment (astonzhang) [inline chapter_gaussian-processes/gp-inference.md:41]: For a full discussion, see MacKay Ch. 28, and Rasmussen and Williams Ch. 5 (! Add references).

Kind reminder on this

**Comment by astonzhang** [comment]
Thanks @andrewgordonwilson for the amazing work! We'll continue to polish it per offline discussions.

**Pull request merged**